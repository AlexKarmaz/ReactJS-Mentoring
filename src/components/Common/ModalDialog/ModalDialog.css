.modalDialog-overlay{
    position: fixed;
    background-color: rgba(0,0,0,0.6);
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    z-index: 1;
}

.modalDialog {
    background-color: #232323;
    color: #ffffff;
    position: fixed;
    top: 10%;
    left: 50%;
    border-radius: 5px;
    transform: translateX(-50%);
    width: 40%;
    overflow: auto;
}

.modalDialog > .closeButton {
    right: 10px;
    top: 10px;
    position: absolute;
}

.modalDialog-title{
    text-transform: uppercase;
    letter-spacing: 4px;
    font-size: 28px;
    font-weight: 100;
    margin: 35 0 0 40;
}

.modalDialog-content {
    margin: 0 40 0 40;
    font-size: 18px;
    font-weight: 100;
    display: flex;
    flex-direction: column;
}
