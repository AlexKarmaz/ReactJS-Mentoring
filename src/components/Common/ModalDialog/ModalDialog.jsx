import React, {useCallback} from 'react';
import CloseButton from '../CloseButton';
import './ModalDialog.css';

const ModalDialog = ({dialogTitle, onDialogClose, children}) => {
    const onClose = useCallback(() => onDialogClose(), [onDialogClose]);

    return (
        <div className='modalDialog'>
            <h2 className='modalDialog-title'>{dialogTitle}</h2>
            <div className='modalDialog-content'>
                {children}
            </div>
            {props.children}
            <CloseButton onClick={onClose} size='medium' />
         </div>
    );
};

export default ModalDialog;
