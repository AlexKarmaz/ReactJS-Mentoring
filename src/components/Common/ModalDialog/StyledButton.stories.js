/* eslint-disable*/
import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import ModalDialog from './ModalDialog';

const wrapperStyle = {
    backgroundColor: '#232323',
};

export default {
    title: 'Modal Dialog',
    component: ModalDialog,
    decorators: [withKnobs],
};

export const Default = () => (
    <div style={wrapperStyle}>
        <ModalDialog onDialogClose={action('ModalDialog-close')} dialogTitle={text('dialogTitle', 'Modal Dialog text')}>
            <div>Dialog body</div>
        </ModalDialog>
    </div>
);
