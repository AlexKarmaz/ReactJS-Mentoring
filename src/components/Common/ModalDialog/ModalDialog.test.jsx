import { shallow, mount } from 'enzyme';
import React from 'react';
import ModalDialog from './ModalDialog';

describe('ModalDialog', () => {
    it('should be rendered correctly', () => {
        const onClose = jest.fn();

        const modalDialog = shallow(<ModalDialog dialogTitle={'test'} onDialogClose={onClose} />);

        expect(modalDialog.find('.modalDialog-title').text()).toEqual('test');
        expect(modalDialog).toMatchSnapshot();
    });

    it('close handler should be called', () => {
        const onClose = jest.fn();

        const modalDialog = mount(<ModalDialog dialogTitle={'test'} onDialogClose={onClose} />);
        modalDialog.find('.closeButton').simulate('click');

        expect(onClose).toBeCalled();
    });
});
