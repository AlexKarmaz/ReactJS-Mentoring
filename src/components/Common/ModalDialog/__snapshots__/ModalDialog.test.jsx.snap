// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog should be rendered correctly 1`] = `ShallowWrapper {}`;
